components/core/src/compressor_frontend/InputBuffer.hpp

```diff
@@ -0,0 +1,82 @@
+#ifndef COMPRESSOR_FRONTEND_INPUT_BUFFER_HPP
+#define COMPRESSOR_FRONTEND_INPUT_BUFFER_HPP
+
+// Project Headers
+#include "../ReaderInterface.hpp"
+#include "Buffer.hpp"
+
+namespace compressor_frontend {
+    class InputBuffer : public Buffer<char> {
+    public:
+
+        /**
+         * Resets input buffer
+         * @return
+         */
+        void reset () override;
+
+        /**
```

Comment:
Comment has too many negatives. How about "Checks if reading into the input buffer is safe in that unconsumed data won't be overwritten"?